iot/iot-tenant-cleaner/src/main/java/io/enmasse/iot/tenantCleaner/infinispan/Application.java

```diff
@@ -0,0 +1,182 @@
+import io.enmasse.iot.service.base.infinispan.cache.DeviceManagementCacheProvider;
+import io.enmasse.iot.service.base.infinispan.config.InfinispanProperties;
+import io.enmasse.iot.service.base.infinispan.devcon.DeviceConnectionKey;
+import io.enmasse.iot.service.base.infinispan.device.DeviceInformation;
+import io.enmasse.iot.service.base.infinispan.device.DeviceKey;
+
+import io.vertx.config.ConfigRetriever;
+import io.vertx.config.ConfigRetrieverOptions;
+import io.vertx.config.ConfigStoreOptions;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import java.util.List;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.Search;
+import org.infinispan.query.dsl.Expression;
+import org.infinispan.query.dsl.Query;
+import org.infinispan.query.dsl.QueryFactory;
+
+public class Application extends AbstractVerticle {
+
+    private static String TENANT_ID_TO_CLEAN;
+
+    private static String INFINISPAN_HOST;
+    private static int INFINISPAN_PORT;
+    private static String INFINISPAN_USERNAME;
+    private static String INFINISPAN_PASSWORD;
+    private static String INFINISPAN_SASL_SERVER;
+    private static String INFINISPAN_SASL_REALM;
+
+    private static RemoteCache<DeviceKey, DeviceInformation> devicesCache;
+    private static RemoteCache<DeviceConnectionKey, String> devicesConnectionCache;
+
+    public Application() {
+    }
+
+    @Override
+    public void start(Future<Void> fut) {
+
+        configure();
+        InfinispanProperties infinispanProperties= getInfinispanProperties();
+
+        DeviceManagementCacheProvider mgmtProvider = null;
+        DeviceConnectionCacheProvider deviceconProvider = null;
+        try {
+            mgmtProvider = new DeviceManagementCacheProvider(infinispanProperties);
+            mgmtProvider.start();
+            devicesCache = mgmtProvider.getDeviceManagementCache();
+
+            deviceconProvider = new DeviceConnectionCacheProvider(infinispanProperties);
+            deviceconProvider.start();
+            devicesConnectionCache = deviceconProvider.getDeviceStateCache();
+        } catch (Exception e){
+            System.err.println(String.format("[ERROR] Unable to access Infinispan caches. {}", e.getMessage()));
+            System.exit(1);
+        }
+
+        // Query and delete entries in devicesCache
+        QueryFactory queryFactory = Search.getQueryFactory(devicesCache);
+        Query query = queryFactory.from(DeviceInformation.class)
+                .select(Expression.property("deviceId"))
+                .having("tenantId")
+                .eq(TENANT_ID_TO_CLEAN)
+                .build();
+
+        List<String[]> matches = query.list();
+        matches.forEach(deviceInfo -> {
+            System.out.println(deviceInfo[0]);
+            DeviceKey key = new DeviceKey(TENANT_ID_TO_CLEAN, deviceInfo[0]);
+            devicesCache.removeAsync(key);
```

Comment:
Calling "async" without waiting for the operation to complete will put in on an internal queue. Having 1.000.000 devices, this would mean a task queue of 1.000.000 async operations, and probably cause some issues.